core/vm/logger.go

```diff
@@ -502,6 +503,13 @@ func (l *StructLogger) MaybeAddFeeRecipientsToStatesAffected(tx *types.Transacti
 	}
 }
 
+func (l *StructLogger) MaybeAddL1BlockInfo(tx *types.Transaction) {
+	if contractAddress := *tx.To(); bytes.Equal(contractAddress[:], types.L1BlockAddr[:]) {
```

Comment:
How about check `tx.IsDepositTx()`?

Reply:
I think it's good to add.

Reply:
Ah, if we check the tx is deposit tx, user-deposited txs will be passed. So it's ok to check the to address is L1Block contract.

Reply:
Currently all txs pass the condition if to address is L1Block, but if `tx.IsDepositTx()` is added, at least deposit txs will pass the condition if to address is L1Block. I thought it might be meaningful, isn't it?

Reply:
Only `0xDeaDDEaDDeAdDeAdDEAdDEaddeAddEAdDEAd0001` can call L1Block contract.